x-pack/plugin/profiling/src/internalClusterTest/java/org/elasticsearch/xpack/profiling/GetStackTracesActionIT.java

```diff
@@ -80,6 +86,70 @@ public void testGetStackTracesFromAPMWithMatch() throws Exception {
         assertEquals("libzip.so", response.getExecutables().get("GXH6S9Nv2Lf0omTz4cH4RA"));
     }
 
+    public void testGetStackTracesFromAPMWithMatchAndDownsampling() throws Exception {
+        TermQueryBuilder query = QueryBuilders.termQuery("transaction.name", "encodeSha1");
+        Index apmTest = resolveIndex("apm-test-001");
+
+        GetStackTracesRequest request = new GetStackTracesRequest(
+            1,
+            1.0d,
+            1.0d,
+            query,
+            "apm-test-*",
+            "transaction.profiler_stack_trace_ids",
+            null,
+            null,
+            null,
+            null,
+            null
+        ) {
+            @Override
+            public boolean equals(Object o) {
+                return super.equals(o);
+            }
+
+            @Override
+            public int hashCode() {
```

Comment:
I'm not fond of this approach but I've started an internal discussion how we can improve the situation when the random sampler aggregation is involved. Nevertheless, I did not want to block progress on the PR.

Reply:
Currently there is https://github.com/elastic/elasticsearch/pull/104830 open to address this. Once this is merged I'd open a small follow-up PR to simplify this test.